docs/codegen/generate-hcl.md

```diff
@@ -125,6 +125,31 @@ Blocks defined at different levels with the same label aren't allowed, resulting
 in failure for the overall code generation process.
 
 
+## Conditional Code Generation
+
+Conditional code generation is achieved by the use of the `condition` attribute.
+The `condition` attribute should always evaluate to a boolean. The file will
+be generated only if it evaluates to **true**.
+
+If the `condition` attribute is absent then it is assumed to be always true.
+
+Any expression that produces a boolean can be used, including references
+to globals and function calls. For example:
+
+```hcl
+generate_hcl "file" {
+  condition = tm_length(global.list) > 0
+  content {
+    locals {
+      list = global.list
+    }
+  }
+}
+```
+
+Will only generate code for stacks that define the global `generate_file` to true.
```

Comment:
i guess this is a mismatch of whatthe condition actually says?

Reply:
I think this is leftover of a change I suggested. @katcipis need to fix here.